shared_model/backend/protobuf/proto_block_factory.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PROTO_BLOCK_FACTORY_HPP
+#define IROHA_PROTO_BLOCK_FACTORY_HPP
+
+#include "backend/protobuf/block.hpp"
+#include "backend/protobuf/empty_block.hpp"
+#include "backend/protobuf/transaction.hpp"
+#include "block.pb.h"
+#include "common/result.hpp"
+#include "interfaces/iroha_internal/unsafe_block_factory.hpp"
+#include "validators/abstract_validator.hpp"
+
+namespace shared_model {
+  namespace proto {
+    /**
+     * ProtoBlockFactory is used to create proto::Block objects
+     */
+    class ProtoBlockFactory : public interface::UnsafeBlockFactory {
+     public:
+      explicit ProtoBlockFactory(
+          std::unique_ptr<shared_model::validation::AbstractValidator<
+              shared_model::interface::BlockVariant>> validator);
+
+      interface::BlockVariant unsafeCreateBlock(
+          interface::types::HeightType height,
+          const interface::types::HashType &prev_hash,
+          interface::types::TimestampType created_time,
+          const interface::types::TransactionsCollectionType &txs) override;
+
+      /**
+       * Create block variant with nonempty block
+       * @return Error if block is empty, or if it is invalid
```

Comment:
Please, add `@param` as well

Reply:
Why? Is the parameter not clear?

Reply:
Clear, but as I understand, we document every parameter and return values